### PR TITLE
cfg-parser: add aliases for yesno

### LINKED ISSUE #906: Additional aliases for yesno
Consider to add some aliases for `yes`/`no` in configuration settings, for example:
- `true`/`false`,
- `enable`/`disable`,
- `enabled`/`disabled`,
- `on`/`off`.

Although the syntax is correctly documented in the guide, this terminology surprised me when I first used `syslog-ng`. Of course `yes`/`no` is also used in config files of other software as well, many (if not all) tools that I've used did provide a subset of the above aliases. Just to let you stay in touch with the real world and to keep in mind the principle of least surprise.

I had this idea after seeing undocumented numerical aliases in #904.

### PR DESCRIPTION
**Pull request by alltilla**
New aliases:
 * true/false
 * enable/disable
 * enabled/disabled

Fixes #906.

Signed-off-by: Attila Szakacs <attila.szakacs@balabit.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
@szemere: Good point, although the fix is easy, I am also not sure if we want this. 

I say, that this change does not remove any previous functionality, and just one sentence in the admin guide could list all the aliases, so this is not a big pain IMO. I can accept closing this PR and #906, though, if we do not want this change.

**Comment by furiel** [comment]
The patch looks good to me! My only concern is an old [comment](https://github.com/balabit/syslog-ng/issues/904#issuecomment-180260148) from @bazsi about keyword collision. I don't entirely understand the comment, nor know it is still relevant, but it would worth a quick sync about it.

**Comment by alltilla** [comment]
Let's discuss this IRL next week, than :) Thanks for the comments!

**Comment by bazsi** [comment]
If we want to support these tokens, they should not be added via keywords. Rather the "yes" "no" rules
should accept an LL_IDENTIFIER and an explicit lookup be used for the potential aliases.

This way if someone has a destination called "enabled" will not cause keyword collision.

I have seen these in the past, the one I remember vividly is a keyword called "null" and I think SUSE using this name as a destination in their default configuration.

**Comment by alltilla** [comment]
Thanks @bazsi, I will do that! :)

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/cfg-parser.c [hunk 1]
```diff
@@ -452,10 +447,52 @@ cfg_process_flag(CfgFlagHandler *handlers, gpointer base, const gchar *flag_)
   return FALSE;
 }
 
+gboolean
+cfg_validate_yesno(const gchar *yesno, gboolean *value)
```
Comment (Kokan) [inline lib/cfg-parser.c]: Have you considered merging the `cfg_validate_yesno` and `cfg_process_yesno` ?

File: lib/cfg-grammar.y [hunk 2]
```diff
@@ -1029,9 +1029,17 @@ string
...
 
 yesno
-	: KW_YES				{ $$ = 1; }
-	| KW_NO					{ $$ = 0; }
-	| LL_NUMBER				{ $$ = $1; }
+  : LL_IDENTIFIER
+    {
+      gboolean value;
+      gboolean success = cfg_validate_yesno($1, &value);
+      CHECK_ERROR((TRUE == success), @1, "Invalid yesno value: %s", $1);
```
[the remark this answers is not in the document]
Comment (alltilla) [reply lib/cfg-grammar.y]: Good idea, I will add this. Thanks!

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/cfg-parser.c (hunk 1 above)
Comment (alltilla) [reply lib/cfg-parser.c]: Yep, I just did that :P

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by lbudai** [comment]
Hmmm... with this patch we just broke all the configs that uses `peer-verify(yes/no)`.
Unfortunately we don't have any tests for this case (not even in the internal testdb...).

The problem is :
`KW_PEER_VERIFY '(' yesno ')' `
`KW_PEER_VERIFY '(' string ')'` ...

and as `yesno` is an `LL_IDENTIFIER`, we got the follwing message from Bison (this was the trigger for me, I've just modified the grammar, and got this error.. but I was sure that I did not make any mistake, so I checked master and found this patch)
```
modules/afsocket/afsocket-grammar.y: warning: 1 reduce/reduce conflict [-Wconflicts-rr]
```


Maybe we should also check bison warning...

I'm going to revert this patch (it's just an old, internally reported, nice-to-have improvement, so we can ship these changes in the next release).

**Comment by alltilla** [comment]
Sure thing, thanks for checking, and sorry for introducing such a bug.
I will make it work in the next release.

**Pull request merged**